### PR TITLE
feat(compute): add compute_reservation_delete 

### PR DESCRIPTION
**Pull request by TetyanaYahodska**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Comment by TetyanaYahodska** [comment]
Created samples for compute_reservation_create and compute_reservation_delete tags. This example shows how to create a reservation for an instance template with a global region

**Comment by TetyanaYahodska** [comment]
Split testing HyperdiskIT.java because ProvisionedIops cannot exceed 10,000 due to a quota limit imposed by Google Cloud Platform (GCP) on your project in the specific region you are working with.

File: compute/cloud-client/src/test/java/compute/reservation/CreateReservationTest.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      return false;
+    }
+  }
+
+  // Create an instance template for the test.
+  @BeforeClass
+  public static void setUp()
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    if (!checkInstanceTemplateExists(PROJECT_ID, INSTANCE_TEMPLATE_NAME)) {
+      CreateInstanceTemplate.createInstanceTemplate(PROJECT_ID, INSTANCE_TEMPLATE_NAME);
```
Comment (Sita04) [inline compute/cloud-client/src/test/java/compute/reservation/CreateReservationTest.java:67]: Same comment. Not sure if the instance template is needed.

File: compute/cloud-client/src/test/java/compute/disks/HyperdiskIT.java
Comment (Sita04) [inline compute/cloud-client/src/test/java/compute/disks/HyperdiskIT.java:1]: Please add only the files related to this sample in this PR. For bug fixes, kindly create a separate PR with the changes.

Comment (Sita04) [reply compute/cloud-client/src/test/java/compute/disks/HyperdiskIT.java:1]: Same comment for other test files.

**Review by Sita04 — CHANGES_REQUESTED** [review]

File: compute/cloud-client/src/main/java/compute/reservation/DeleteReservation.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package compute.reservation;
+
+// [START compute_reservation_delete]
```
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/reservation/DeleteReservation.java:19]: The same Reservations Client is used to delete local and regional reservations. I suppose we can left it as is.

File: compute/cloud-client/src/test/java/compute/disks/HyperdiskIT.java
Comment (TetyanaYahodska) [reply compute/cloud-client/src/test/java/compute/disks/HyperdiskIT.java:1]: Thank you! Created new PR for bug fixing.

File: compute/cloud-client/src/test/java/compute/reservation/CreateReservationTest.java (hunk 1 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/test/java/compute/reservation/CreateReservationTest.java]: You are right. Fixed code and moved to another PR.

File: compute/cloud-client/src/main/java/compute/reservation/CreateReservation.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    String reservationName = "YOUR_RESERVATION_NAME";
+    // Name of your instance template.
+    String instanceTemplateName = "INSTANCE_TEMPLATE_NAME";
+    // The number of virtual machines you want to create.
+    long numberOfVms = 3;
+
+    createReservation(projectId, reservationName, instanceTemplateName, numberOfVms, zone);
+  }
+
+  // Creates a reservation in a project for the instance template for the global region.
```
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/reservation/CreateReservation.java]: You are right. Fixed code and moved to another PR.

File: compute/cloud-client/src/main/java/compute/reservation/CreateReservation.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+public class CreateReservation {
+
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Cloud project you want to use.
+    String projectId = "YOUR_PROJECT_ID";
+    // Name of the zone in which you want to create the disk.
+    String zone = "us-central1-a";
+    // Name of the disk you want to create.
```
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/reservation/CreateReservation.java]: You are right. Fixed comment

File: compute/cloud-client/src/main/java/compute/reservation/CreateReservation.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the `reservationsClient.close()` method on the client to safely
+    // clean up any remaining background resources.
+    try (ReservationsClient reservationsClient = ReservationsClient.create();
+         InstanceTemplatesClient templatesClient = InstanceTemplatesClient.create()) {
+
+      // Get the instance template.
+      InstanceTemplate instanceTemplate = templatesClient.get(projectId, instanceTemplateName);
```
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/reservation/CreateReservation.java]: Created two samples for global and regional location and moved to another PR

File: compute/cloud-client/src/main/java/compute/reservation/DeleteReservation.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class DeleteReservation {
+
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // projectId: project ID or project number of the Cloud project you want to use.
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/reservation/DeleteReservation.java:34]:
```suggestion
    // Project ID or project number of the Cloud project you want to use.
```

File: compute/cloud-client/src/test/java/compute/disks/HypedisksIT.java [hunk 7]
```diff
@@ -132,4 +132,4 @@ public void stage2_CreateHyperdiskStoragePoolTest()
     Assert.assertTrue(disk.getType().contains("hyperdisk-balanced"));
     Assert.assertTrue(disk.getZone().contains(ZONE));
   }
-}
+}
```
Comment (Sita04) [inline compute/cloud-client/src/test/java/compute/disks/HypedisksIT.java:135]: Please move this change to the other PR with hyperdisk samples.

**Review by Sita04 — CHANGES_REQUESTED** [review]
Kindly include the test for `DeleteReservation` sample in this PR.

**Comment by TetyanaYahodska** [comment]
In the next PRs, I will call the deleteReservation method in the cleanup() methods and add an assertion to it. Now it is manually tested and it works as expected.

File: compute/cloud-client/src/main/java/compute/reservation/DeleteReservation.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/reservation/DeleteReservation.java:19]: I used this method to delete reservation for instance Template with global location and regional location and it works as expected. I suppose this tag naming is short and clear.

**Comment by TetyanaYahodska** [comment]
I changed the zones for HyperdisksIT due to some limitations of Google Cloud regarding the Hyperdisk storage pool created for a single zone

File: compute/cloud-client/src/main/java/compute/reservation/CreateReservation.java [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+      // Create the reservation.
+      Reservation reservation =
+          Reservation.newBuilder()
+              .setName(reservationName)
+              .setZone(zone)
```
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/reservation/CreateReservation.java]: This code fixed and moved to another PR

File: compute/cloud-client/src/main/java/compute/reservation/CreateReservation.java [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+              .setSpecificReservation(
+                  AllocationSpecificSKUReservation.newBuilder()
+                      // Set the number of instances
+                      .setCount(numberOfVms)
+                      // Set instance properties
+                      .setInstanceProperties(
+                          AllocationSpecificSKUAllocationReservedInstanceProperties.newBuilder()
+                              .setMachineType("n1-standard-1")
```
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/reservation/CreateReservation.java]: Following Google documentation we can set MachineType for Reservation too. gcloud compute reservations create RESERVATION_NAME \
    --machine-type=MACHINE_TYPE \
    --require-specific-reservation \
    --vm-count=NUMBER_OF_VMS \
    --zone=ZONE

**Review by Sita04 — APPROVED** [review]

File: compute/cloud-client/src/test/java/compute/reservation/DeleteReservationIT.java
Comment (Sita04) [inline compute/cloud-client/src/test/java/compute/reservation/DeleteReservationIT.java:1]: Can you call the file just ReservationIT? 
IIUC, you will be including Create Reservation and other related samples in this file, right?

**Review by Sita04 — COMMENTED** [review]
Please address the comment before merging.

File: compute/cloud-client/src/test/java/compute/reservation/DeleteReservationIT.java
Comment (TetyanaYahodska) [reply compute/cloud-client/src/test/java/compute/reservation/DeleteReservationIT.java:1]: Thank you! Done

**Pull request merged**